src/Uno.UI/UI/Xaml/Controls/Border/Border.cs

```diff
@@ -322,17 +322,7 @@ protected override void OnBackgroundChanged(DependencyPropertyChangedEventArgs e
 
 	internal override bool CanHaveChildren() => true;
 
-	internal override bool IsViewHit() => IsViewHitImpl(this);
-
-	internal static bool IsViewHitImpl(FrameworkElement element)
-	{
-		_Debug.Assert(element is Panel
-			|| element is Border
-			|| element is ContentPresenter
-		);
-
-		return element.Background != null;
-	}
+	internal override bool IsViewHit() => ((IBorderInfoProvider)this).IsViewHit();
```

Comment:
This may change behavior on non-Skia. It was already not correct behavior, but now will be incorrect in a different way, I think.

Reply:
We could guard this with `#if __SKIA__`, but the change makes sense in general I think, so I would say it's better to change the behaviour on non-skia targets even if it's still not accurate.